### PR TITLE
Edit Page Question

### PR DESCRIPTION
**Pull request by mpeels**
## Description

The original `Edit question` workflow targeted the `Wa_question` entity. When editing a question from within the context of a page, the `Wa_ui_metadata` entity should be targeted, not the "base" `Wa_question`.

This PR adds the ability to edit questions within the context of a page and adds the UI components to carry out that task.

Additional APIs needed to be created:
1. allow editing of text, numeric, date, and coded questions.
2. fetch additional data required to perform editing of questions
3. Validate data mart column name within the context of a page

## Tickets

* [CNFT2-1571](https://cdc-nbs.atlassian.net/browse/CNFT2-1571)

### Editing a Text question
UpdateText

### Editing a Numeric question
updateNumeric

### Editing a Date question
updateDate

### Editing a Coded question
updateCoded

## Checklist before requesting a review
- [ ] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [ ] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [ ] All code is covered by unit or feature tests

[CNFT2-1571]: https://cdc-nbs.atlassian.net/browse/CNFT2-1571?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by lramakr — APPROVED** [review]

File: apps/modernization-ui/src/apps/page-builder/generated/services/QuestionControllerService.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ export class QuestionControllerService {
      * @returns any Created
      * @throws ApiError
      */
-    public static updateCodedQuestionUsingPut({
+    public static updateCodedQuestionUsingPut1({
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/page-builder/generated/services/QuestionControllerService.ts:99]: this is unfortunate 😭

File: apps/modernization-ui/src/apps/page-builder/page/management/edit/edit-question/EditQuestionModal.scss
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/page-builder/page/management/edit/edit-question/EditQuestionModal.scss:1]: question:  Was this file intended to be removed?  It looks like it's the same as `edit-question-modal.module.scss`

File: apps/modernization-ui/src/apps/page-builder/page/management/edit/edit-question/form/EditFields.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                    <>
+                        <label htmlFor="required" className={styles.toggleLabel}>
+                            Required? <span className={styles.mandatory}>*</span>
+                        </label>
+                        <div className={styles.toggleGroup}>
+                            <div>Not required</div>
+                            <ToggleButton checked={value === true} name="required" onChange={onChange} />
+                            <div>Required</div>
+                        </div>
+                    </>
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/page-builder/page/management/edit/edit-question/form/EditFields.tsx:55]: question (for the future):  Should we target this pattern as a re-usable component?

File: apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/component/ContentNode.java [hunk 3]
```diff
@@ -23,10 +23,7 @@ public record Attributes(
       String dataMartColumnName,
       int nbsComponentId,
       String dataLocation,
-      String publishIndicator
-  ) {
-
-
+      boolean isPublished) {
```
Comment (adamloup-enquizit) [inline apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/component/ContentNode.java:26]: ❤️

File: apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/question/request/update/UpdateDateQuestionRequest.java [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package gov.cdc.nbs.questionbank.question.request.update;
 
+import gov.cdc.nbs.questionbank.question.request.create.CreateDateQuestionRequest.DateMask;
```
Comment (adamloup-enquizit) [inline apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/question/request/update/UpdateDateQuestionRequest.java:3]: question (non-blocking): What is `DateMask`?  If there is only one is it just a boolean?

File: apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/question/request/update/UpdateNumericQuestionRequest.java [hunk 5]
```diff
@@ -1,5 +1,6 @@
 package gov.cdc.nbs.questionbank.question.request.update;
 
+import gov.cdc.nbs.questionbank.question.request.create.CreateNumericQuestionRequest.NumericMask;
```
Comment (adamloup-enquizit) [inline apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/question/request/update/UpdateNumericQuestionRequest.java:3]: question (non-blocking): Should `NumericaMask` be a standalone enum?

File: apps/question-bank/src/test/java/gov/cdc/nbs/questionbank/entity/WaUiMetadataTest.java [hunk 6]
```diff
@@ -12,271 +13,709 @@
...
+import gov.cdc.nbs.questionbank.page.command.PageContentCommand.UpdateDateQuestion;
+import gov.cdc.nbs.questionbank.page.command.PageContentCommand.UpdateNumericQuestion;
+import gov.cdc.nbs.questionbank.page.command.PageContentCommand.UpdateTextQuestion;
+import gov.cdc.nbs.questionbank.page.content.PageContentModificationException;
 import gov.cdc.nbs.questionbank.page.exception.AddQuestionException;
 import gov.cdc.nbs.questionbank.question.command.QuestionCommand.Update;
+import gov.cdc.nbs.questionbank.question.request.QuestionRequest.ReportingInfo;
 import gov.cdc.nbs.questionbank.support.QuestionEntityMother;
 
 @ExtendWith(MockitoExtension.class)
```
Comment (adamloup-enquizit) [inline apps/question-bank/src/test/java/gov/cdc/nbs/questionbank/entity/WaUiMetadataTest.java:27]: nit:  I don't think this is doing anything.

File: apps/modernization-ui/src/apps/page-builder/components/FormDivider/HorizontalRule.tsx [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import styles from './horizontal-rule.module.scss';
+
+export const HorizontalRule = () => {
+    return <div className={styles.hr} />;
```
[the remark this answers is not in the document]
Comment (mpeels) [reply apps/modernization-ui/src/apps/page-builder/components/FormDivider/HorizontalRule.tsx:4]: for now, yes. The only use is within the Edit / Create question form as a `separator`

image

File: apps/question-bank/src/test/java/gov/cdc/nbs/questionbank/entity/WaUiMetadataTest.java (hunk 6 above)
Comment (mpeels) [reply apps/question-bank/src/test/java/gov/cdc/nbs/questionbank/entity/WaUiMetadataTest.java]: you sir, are correct

**Review by alaapbharadwaj — APPROVED** [review]

**Review by lramakr — APPROVED** [review]
